### PR TITLE
Update and rename tests.yml to CI.yml

### PR DESCRIPTION
**Pull request by offtherailz**
## Description

This PR adds to the CI process the deploy of maven packages. 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [x] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

[#geosolutions-it/internal/323
](https://github.com/geosolutions-it/internal/issues/343)
**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information

In order to test it, I configured my repo to use this also in a PR, that was successful (adding proper credentials)

See this: https://github.com/offtherailz/MapStore2/pull/35

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -147,7 +147,56 @@ jobs:
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: '8.x'
+      ############
+      # Build
+      ##########
+      - run : "npm install"
+      - run: "npm run fe:build"
+      - run: "npm run jsdoc:build"
+      - run: "mvn clean install -Dmapstore2.version=$GITHUB_REF_NAME -Pprintingbundle"
```
Comment (offtherailz) [inline .github/workflows/CI.yml:171]:
```suggestion
      - name: "setting up Java"
        uses: actions/setup-java@v1
        with:
          java-version: '11.x'
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -147,7 +147,49 @@ jobs:
...
+        env:
+          MAVEN_USERNAME: ${{ secrets.GS_MAVEN_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.GS_MAVEN_PASSWORD }}
 
-
+      - name: Publish package on Gs maven repository
+        run: mvn -B clean install -Dmapstore2.version=$GITHUB_REF_NAME deploy
+        env:
+          MAVEN_USERNAME: ${{ secrets.GS_MAVEN_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.GS_MAVEN_PASSWORD }}
```
Comment (offtherailz) [inline .github/workflows/CI.yml:193]:
```suggestion
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -147,7 +147,49 @@ jobs:
...
+                printing/target/mapstore-printing.zip  
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          server-id: geosolutions
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: Publish package
```
Comment (offtherailz) [inline .github/workflows/CI.yml:183]:
```suggestion
      - name: Publish maven packages
```

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -147,7 +147,49 @@ jobs:
...
+        run: mvn -B clean install deploy
```
Comment (offtherailz) [inline .github/workflows/CI.yml:184]:
```suggestion
        run: mvn install deploy --projects \!product
```

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -147,7 +147,44 @@ jobs:
...
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          server-id: geosolutions
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: Publish maven packages
+        run: mvn install deploy --projects \!product
```
Comment (offtherailz) [inline .github/workflows/CI.yml:184]:
```suggestion
        run: mvn clean install deploy -f java/pom.xml`
```

File: .github/workflows/CI.yml [hunk 6]
```diff
@@ -147,7 +147,37 @@ jobs:
...
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          server-id: geosolutions
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: Publish maven packages
+        run: mvn clean install deploy -f java/pom.xml
```
Comment (tdipisa) [inline .github/workflows/CI.yml:177]: why not only java/services/pom.xml?

**Review by tdipisa — APPROVED** [review]

**Pull request merged**